pymc3/step_methods/sgmcmc.py

```diff
@@ -298,3 +311,98 @@ def competence(var, has_grad):
         if var.dtype in continuous_types and has_grad:
             return Competence.COMPATIBLE
         return Competence.INCOMPATIBLE
+
+
+class ConstantStochasticGradient(BaseStochasticGradient):
+    R"""
+    ConstantStochasticGradient: approximates stochastic variational inference
+    
+    It approximates stochastic variational inference
+    where it differs from SGFS and many other MCMC techniques which
+    converge towards the exact posterior. In comparison to SGFS the paper 
+    discusses a proof for the optimal preconditioning matrix
+    based on variational inference, so there is no additional input.
+     
+    Parameters
+    -----
+    vars : list
+        model variables
+
+    References
+    -----
+    -   Stochastic Gradient Descent as Approximate Bayesian Inference
+        https://arxiv.org/pdf/1704.04289v1.pdf
+    """
+
+    def __init__(self, vars=None, **kwargs):
+        """
+        Parameters
+        ----------
+        vars : list 
+            Theano variables, default continuous vars
+        kwargs: passed to BaseHMC
+        """
+        super(ConstantStochasticGradient, self).__init__(vars, **kwargs)
```

Comment:
Should add an experimental warning.

Reply:
The experimental warning is present in this line https://github.com/shkr/pymc3/blob/45a45ab0f78b480b8accb27f168164bb213cd280/pymc3/step_methods/sgmcmc.py#L112